### ISSUE TITLE
Grad should be a Value instead of python/numpy scalar

### ISSUE THREAD

**Issue by MarisaKirisame** [karpathy/micrograd#3]
So you can do higher order autodiff.

**Comment by karpathy** [comment]
yes, similar to PyTorch, a good feature request. I'll see if I can find a way to do it without sacrificing  the clarity, cleanliness and compactness.

**Comment by zvikiNozadze** [comment]
I can gratfuly contribute if @karpathy wants, I ve built similar project (autograd on github).

**Comment by evcu** [comment]
I played with higher-order grads a bit in my fork. It was this plus some small changes. Created a pr: #8

**Comment by ghltshubh** [comment]
I think the point is not to make another framework but pedagogy
